header-rewriter/tests/recursion/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+if(LIBIA2_INSECURE)
+    set(STDOUT_REF ${CMAKE_CURRENT_SOURCE_DIR}/Output/recursion.insecure.out)
+else()
+    set(STDOUT_REF ${CMAKE_CURRENT_SOURCE_DIR}/Output/recursion.out)
+endif()
+configure_file(${STDOUT_REF} ${CMAKE_CURRENT_BINARY_DIR}/recursion.out)
+
+# Build the wrapper lib for the recursion
+define_ia2_wrapper(
+    HEADERS recursion_dso.h
+    COMPARTMENT_PKEY 2
+    INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include/dso/
+    CALLER_PKEY 1
+)
+
+# Build the wrapper lib for the main binary
+define_ia2_wrapper(
+    WRAP_MAIN
+    COMPARTMENT_PKEY 1
+    HEADERS recursion_main.h
+    INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include/main/
+    CALLER_PKEY 2
+)
+
+# Build the recursion lib
+define_shared_lib(
+    SRCS dso.c
+    LINK_LIBS recursion-main-wrapper
+    LINK_OPTS "-Wl,-T${libia2_BINARY_DIR}/padding.ld"
```

Comment:
This doesn't need to block merging, but I've forgotten: what exactly requires the padding linker script? Some of our tests define shared libs with a `LINK_OPTS` entry for it (mostly ones that initialize two compartments?) and others don't; cf. `ro_sharing` (does) and `tls_protected` (doesn't).

Reply:
afaik it's only needed to align shared globals to isolated pages.